hugo/content/guides/keywords-as-identifiers/_index.md

```diff
@@ -0,0 +1,157 @@
+
+After creating the semantic token provider, you need to register the `HelloWorldSemanticTokenProvider` in `hello-world-module.ts` in the following way:
+
+```ts
+export const HelloWorldModule: Module<HelloWorldServices, PartialLangiumServices & HelloWorldAddedServices> = {
+    lsp: {
+        SemanticTokenProvider: (services) => new HelloWorldSemanticTokenProvider(services)
+    },
+    validation: {
+        HelloWorldValidator: () => new HelloWorldValidator()
+    }
+};
+```
+
+Now rebuild and restart your application and test the improvements of the second step:
+
+![screenshot with fixed semantic token](fixed-2-token.png)
+
+The `HelloWorldSemanticTokenProvider` works, and you might see a different highlighting XOR you might not see any difference, e.g. "Hello" is still blue here. This leads us to the third step.
+
+The __third step__ is to ensure, that your editor supports the assigned semantic tokens:
+Depending on your editor and the currently selected color theme, the semantic token type selected in `HelloWorldSemanticTokenProvider` might not be supported or didn't got a different color in the color theme.
+The easiest way to detect such problems is to change the current color theme and to try some others.
+Note, that VS Code allows to switch off semantic highlighting for all themes with the setting `editor.semanticHighlighting.enabled`.
+
+After switching from "Dark (Visual Studio)" to "Dark Modern" in VS Code, the example looks as expected.
+You can switch the current color theme in VS Code with `cmd + K` `cmd + T` (or via the menu: Code -> Settings... -> Theme -> Color Theme).
+
+![screenshot with supporting color theme](fixed-3-style-2.png)
+
+Now "Hello" is marked in purple, when it is used as keyword, and in green, when it is used as value for the name of a person.
+Another solution is to select a different semantic type for your token in step two.
+A more elaborate solution is to create your own color theme and to ship it with your VS Code extension of your language.
+
+While __step one__ is mandatory to enable keywords as values in general,
+__step two__ improves the user experience of your language.
+While step one and step two can be handled in the LSP server once for your language, __step three__ highly depends on your editor and its color themes (in the LSP clients), which makes step three quite complicated to handle.
+
+
+Now you know, how to technically enable keywords as regular values for properties.
+As home work, it is your task to enable the keyword "person" as name for persons in the example.
+Whether it makes sense to support keywords as values in your language at all is up to you to discuss with the users of your language!
+
+
+Some hints beyond this guide:
+
+- In [multi-grammar projects](/guides/multiple-languages), only keywords of the included grammars are affected by this general problem, but not keywords of other languages or Langium grammar files.
```

Comment:
Is there a way to find out about all keywords? I think we have a constant somewhere ins the `ast.ts`, but I am not very sure about this.

Reply:
That is a good improvement, thank! I added a hint to the generated TextMate grammar.